Implemented MyLinkedList with basic linked list functionality:

Mwxyz2022 opened the pull request:
- add, add at index, addAll methods
- get, set, remove by index and by value
- size and isEmpty methods
- internal Node class with double links (prev and next)
- index validation and efficient node retrieval from head or tail

Mwxyz2022 committed:
Implemented MyLinkedList with basic linked list functionality:
- add, add at index, addAll methods
- get, set, remove by index and by value
- size and isEmpty methods
- internal Node class with double links (prev and next)
- index validation and efficient node retrieval from head or tail

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -1,47 +1,167 @@
-        return null;
+        checkIndex(index);
+        Node<T> nodeToRemove = getNodeByIndex(index);
+        removeNode(nodeToRemove);
+        return nodeToRemove.value;
```
Issue: You are returning `nodeToRemove.value` after calling `removeNode(nodeToRemove)`. If `removeNode` modifies or clears the node's value, this will return an incorrect value. According to standard practice and the checklist, you should save the value in a variable before removing the node, then return it.

mateacademy-ai-mentor commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -1,47 +1,167 @@
+
+        while (current != null) {
+            if (Objects.equals(current.value, object)) {
+                removeNode(current);
+                return true;
```
Issue: In the `remove(T object)` method, you return `true` after calling `removeNode(current)`, but if you ever need to return the removed value (for consistency with `remove(int)`), you should save the value before removal. While this method returns a boolean, ensure that the node's value is not needed after removal.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your hard work! 😊 There is a critical issue in your implementation: in both `remove(int index)` and `remove(T object)` methods, you return the node's value after calling `removeNode`, which may result in returning an incorrect value if the node is modified or cleared during removal. According to standard practice and the checklist, you should save the value of the node to a variable **before** removing it, and then return this saved value. Please fix this to ensure your code meets the core requirements and behaves correctly. Keep going—you’re almost there! 🚀

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

Mwxyz2022 committed:
Refactoring:
- save value before removing node to return correct data.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job! 🎉 Your implementation now fully meets all the core requirements and constraints of the task. All standard list operations are correctly implemented, edge cases are handled, and your remove methods save and return the correct value as required. There are no outstanding issues, so you can confidently move forward. Keep up the excellent work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

Elena-Bruyako commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -1,47 +1,168 @@
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
```
remove

Elena-Bruyako commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -1,47 +1,168 @@
+
+    }
+
+    private static class Node<T> {
```
[inner class should be placed at the end of outer class](https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md#make-nodet-an-inner-class)

Elena-Bruyako commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -1,47 +1,168 @@
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T value) {
```
[When creating a new Node<T>, pass all the fields to the Node constructor.](https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md#when-creating-a-new-nodet-pass-all-the-fields-to-the-node-constructor)

Elena-Bruyako commented on src/main/java/core/basesyntax/MyLinkedList.java:
```diff
@@ -1,47 +1,168 @@
+        Node<T> current = head;
+
+        while (current != null) {
+            if (Objects.equals(current.value, object)) {
```
[Don't use class Objects.](https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md#dont-use-class-objects)

Mwxyz2022 committed:
Refactoring:
 - changed Node class to set next and prev links when creating a node, making it simpler.
 - simplified adding elements to the end of the list by setting the new node's prev to the current tail.
- made inserting elements at an index cleaner by using the updated Node setup.
- replaced the old remove method with a clearer one called unlink.

Mwxyz2022 committed:
fixed syntax
